src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -532,6 +569,82 @@ cJSON *wm_vuldet_decode_advisories(char *advisories) {
     return json_advisories;
 }
 
+int wm_vuldet_compare_pkg(cve_vuln_pkg *Pkg1, cve_vuln_pkg *Pkg2) {
+    if (!(Pkg1->src_name && Pkg2->src_name &&
+          !strcmp(Pkg1->src_name, Pkg2->src_name))) // Source name
+    {
+        return -1;
+    }
+
+    if (!(Pkg1->bin_name && Pkg2->bin_name &&
+          !strcmp(Pkg1->bin_name, Pkg2->bin_name))) // Binary name
+    {
+        return -1;
+    }
+
+    if (!(Pkg1->version && Pkg2->version &&
+          !strcmp(Pkg1->version, Pkg2->version))) // Version
+    {
+        return -1;
+    }
+
+    if (!(Pkg1->arch && Pkg2->arch &&
+          !strcmp(Pkg1->arch, Pkg2->arch))) // Archiquitecture
+    {
+        return -1;
+    }
+
+    return 0;
+}
+
+    int wm_vuldet_add_cve_node(cve_vuln_pkg *newPkg, const char *cve, OSHash *cve_table)
+{
+    int status = 0;
+    cve_vuln_pkg *lastPkg = NULL;
+    cve_vuln_pkg *tmpPkg  = NULL;
+
+    if (status = OSHash_Add(cve_table, cve, (void *)newPkg), !status) {
```

Comment:
How can we know if the hash table is full when adding a new item? Could we detect that status to call `OSHash_setSize()`  and resize the table?

Reply:
The current hash table implementation supports an unlimited number of nodes by using a linked list per table entry, where several nodes can share the same one in case of hash collision. This solution has a  performance hit though, so it must be carefully used.